mycore-base/src/main/java/org/mycore/common/xml/MCRXMLFunctions.java

```diff
@@ -1031,4 +1035,48 @@ public static String toNCNameSecondPart(String name) {
         return name.replaceAll("[^\\w\\-\\.]*", "");
     }
 
+    /**
+     * This only works with text nodes
+     * @param nodes
+     * @return
+     */
+    public static NodeList distinctValues(NodeList nodes) {
+        SortedSet<Node> set = new TreeSet<>(new Comparator<Node>() {
+            @Override public int compare(Node node, Node t1) {
+
+                String nodeValue = node.getNodeValue();
+                String nodeValue1 = t1.getNodeValue();
+
+                if (nodeValue != null && nodeValue1 != null) {
+                    return nodeValue.equals(nodeValue1) ? 0 : -1;
```

Comment:
skip to many if clauses:
```suggestion
                    return (nodeValue == nodeValue1 || nodeValue.equals(nodeValue1)) ? 0 : -1;
```